### DISCUSSION TITLE
Question emqx with node.js server ko

### DISCUSSION THREAD

**Discussion by qiwalker** [emqx/emqx#5212] [Q&A]
hi

kindly advice
is it possible to use mini pc Minix z83 - 4 plus (windows10) for Emqx broker for handle 100 connection and node.js server ?

thanks

**Comment by k32** [comment]
Hello,

The resource footprint of a single connection in EMQ X is negligible, what actually matters is the amount of data transferred over the connections.

**Comment by qiwalker** [comment]
hi 
thx for promp reply
my node using pubsub and max buffer is 50,
publish data is MacAddress+temperature value and data publish every 3 second

**Comment by qzhuyan** [comment]
@qiwalker you also need to consider the number of subscribers and publishers. 

>publish data is MacAddress+temperature value and data publish every 3 second
is it from one connection or from all the connections? 

BTW, we don't know what runs in your nodejs so we cannot estimate.

**Comment by qiwalker** [comment]
i use 100 pcs esp8266 as node to get data from thermocouple, and send data to Node.js server
each esp8266 will send data every 3 second to EMQX Broker.

to handle MQTT communication to node.js server i use mqtt.js
from node.js server using socket.io to show real time data to web application
the web application also use node.js for server.

**Comment by k32** [comment]
I emulated this scenario using emqtt_bench (100 publishers that push 256-bytes of data every 3 seconds and 10 subscribers to the topic), and the EMQX process is taking ~150MB of RAM and 2% CPU.
Please note that I ran this test on Linux, since I don't have a windows machine at hand, but I believe there should not be much difference.
I hope this helps.

**Reply by qiwalker** [reply]
hi k32, 

thankyou so much, 
anyway what is 300% CPU ?,  on windows max value for CPU is only 100%

below windows task manager show cpu percentage

![image](https://user-images.githubusercontent.com/45870169/125265390-feeedd80-e32e-11eb-81fc-89ba9b022e27.png)

**Reply by k32** [reply]
300% CPU load means the process fully utilizes 3 CPU cores.

**Reply by k32** [reply]
P.S. Apologies, I had to edit my answer because I ran benchmark where messages were published every 3 milliseconds rather than seconds. Please see the updated answer.

**Reply by qiwalker** [reply]
that's good i think this mini PC should be able to use for EMQX and Node.js
thankyou so much for your support, 

one more question, do i need to run script on command line "emqx.cmd start" when pc is restart ?

for know when pc is power on or restart i must type "emqx.cmd start", if i not do this when open the web dashboard and do login the web show "No Network"

regards.
walker

**Reply by k32** [reply]
> one more question, do i need to run script on command line "emqx.cmd start" when pc is restart ?

Yes, I believe so.
